File: Web_Visualization/src/test/java/com/gatech/project/dao/LanguageSearchCountDAOTest.java
package com.gatech.project.dao;

import com.gatech.project.domain.LanguageSearchCount;
import com.gatech.project.params.Params;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import java.util.List;

@SpringBootTest
class LanguageSearchCountDAOTest {
    @Test
    public void LanguageSearchCountDAOTest() {
        String tableName = Params.TABLENAME;
        Long check = new Long(3);
        LanguageSearchCountDAO dao = new LanguageSearchCountDAO();
        List<LanguageSearchCount> list = dao.query(tableName, Params.ROWKEY_TEST);
        for (LanguageSearchCount model : list) {
            System.out.println(model.getName() + " : " + model.getValue());
            if(model.getName() == "20200328") {
                assert(model.getValue() == check);
            }
        }
    }
}

File: Logs_Analysis/src/test/scala/com/gatech/project/dao/LanguageSearchCountDAOTest.scala
package com.gatech.project.dao

import org.junit._
import Assert._
import com.gatech.project.domain.LanguageSearchCount

import scala.collection.mutable.ListBuffer

@Test
class LanguageSearchCountDAOTest {

  @Test
  def LanguageSearchCountDAOTest() = {
    val list = new ListBuffer[LanguageSearchCount]
    list.append(LanguageSearchCount("20200111_7",8))
    list.append(LanguageSearchCount("20200111_8",9))
    list.append(LanguageSearchCount("20200111_9",100))

    LanguageSearchCountDAO.saveOnce(list)

    assert(LanguageSearchCountDAO.count("20200111_7") == 8.toLong)
    assert(LanguageSearchCountDAO.count("20200111_8") == 9.toLong)
    assert(LanguageSearchCountDAO.count("20200111_9") == 100.toLong)

    val list2 = new ListBuffer[LanguageSearchCount]
    list2.append(LanguageSearchCount("20200111_7",8))
    list2.append(LanguageSearchCount("20200111_8",9))
    list2.append(LanguageSearchCount("20200111_9",100))
    LanguageSearchCountDAO.save(list2)
    assert(LanguageSearchCountDAO.count("20200111_7") == 16.toLong)
    assert(LanguageSearchCountDAO.count("20200111_8") == 18.toLong)
    assert(LanguageSearchCountDAO.count("20200111_9") == 200.toLong)


  }

}
